### DISCUSSION TITLE
Mask w, Mask h

### DISCUSSION THREAD

**Discussion by andupotorac** [Bing-su/adetailer#137] [Ideas]
We have a Mask offset for X and Y, which allows us to control the point where the mask starts. But we don't have any control over where the mask should end, and it ends at the top and on the right of the image.

What if instead, we could also control the width and height of the mask area? This way we could precisely control the part of the image for which we want faces/hands changed/fixed.

**Comment by Bing-su** [comment]
Good idea, but do you have any ideas on how to apply the x and y axis of the segmentation model?

**Reply by andupotorac** [reply]
I thought this works like this, where you basically set the area (x, y, w, h) where the segmentation happens. So then it will find only the face & hands you're interested in. 
<img width="634" alt="image" src="https://github.com/Bing-su/adetailer/assets/3269738/8c7b8db5-3564-476a-bb69-e19d20a39889">

**Comment by Bing-su** [comment]
Considering the segmentation model, I think this would be better handled with the mask erosion/dilation feature.
